deploy/cloud/helm/network-config-wizard.sh

```
@@ -97,10 +97,20 @@ if kubectl get crd gateways.networking.istio.io &>/dev/null; then
-    jq -r '.items[].spec.hosts[]?' | grep -v null | grep -v '\*' | \
-    while read host; do extract_domain_suffix "$host"; done | sort | uniq))
+  HOST_PATTERNS_RAW=$(echo "$VIRTUALSVC_JSON" | jq -r '.items[].spec.hosts[]?' 2>/dev/null | grep -v null | grep -v '\*' || true)
+
+  HOST_PATTERNS=()
+  if [[ -n "$HOST_PATTERNS_RAW" ]]; then
+    while read -r host; do
+      suffix=$(extract_domain_suffix "$host")
+      HOST_PATTERNS+=("$suffix")
+    done <<< "$HOST_PATTERNS_RAW"
+
+    HOST_PATTERNS=($(printf "%s\n" "${HOST_PATTERNS[@]}" | sort | uniq))
```

Shellcheck (0.10.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)